plugins/obs-webrtc/whip-output.cpp

```diff
@@ -148,16 +140,32 @@ void WHIPOutput::ConfigureVideoTrack(std::string media_stream_id,
 		ssrc, cname, video_payload_type,
 		rtc::H264RtpPacketizer::defaultClockRate);
 
-	if (is_av1) {
+	const obs_encoder_t *encoder = obs_output_get_video_encoder2(output, 0);
+	if (encoder == nullptr) {
+		return;
+	}
+
+	const char *codec = obs_encoder_get_codec(encoder);
+	if (strcmp("h264", codec) == 0) {
+		video_description.addH264Codec(video_payload_type);
+		packetizer = std::make_shared<rtc::H264RtpPacketizer>(
+			rtc::H264RtpPacketizer::Separator::StartSequence,
+			rtp_config, MAX_VIDEO_FRAGMENT_SIZE);
+#if ENABLE_HEVC
+	} else if (strcmp("hevc", codec) == 0) {
+		video_description.addH265Codec(video_payload_type);
+		packetizer = std::make_shared<rtc::H265RtpPacketizer>(
+			rtc::H265RtpPacketizer::Separator::StartSequence,
+			rtp_config, MAX_VIDEO_FRAGMENT_SIZE);
+#endif
+	} else if (strcmp("av1", codec) == 0) {
```

Comment:
me neither, strcmp was on the original code, so just followed the same pattern for adding  hevc. If there is a getter returning an enum I am happy to change it to a "switch"

Reply:
I don't think this output has any available codec enum or getter for it, which is why I suggested that this is perhaps something to clean up later.

Reply:
but the `codec` is gathered from the encoder, not from the output, so even we had an enum and a codec name mapping, we would still need an `strcmp`, right?

Reply:
I believe this is currently the best method to implement this code. I also believe that we should have use a codec enum instead of string ids originally in the API. Too bad, but this works fine enough